src/java/picard/illumina/ClusterDataToSamConverter.java

```diff
@@ -134,12 +169,29 @@ private SAMRecord createSamRecord(final ReadData readData, final String readName
             sam.setAttribute(SAMTag.BC.name(), unmatchedBarcode);
         }
 
-        if (molecularIndex != null) {
+        if (!molecularIndexes.isEmpty()) {
             //TODO: add RX and QX to the list of SAMTags and change this. initial discussion
```

Comment:
the todos here become very out of place..coudl you move them to where I commented above instead?